addons/api/addon/adapters/application.js

```diff
@@ -287,17 +287,23 @@ export default class ApplicationAdapter extends RESTAdapter.extend(
       detail: payload?.message,
       source: { pointer: '/data' },
     };
+    // NOTE: When translating API attributes to Ember Data attributes, we need
+    // to add a check here to ensure that the translation is handled correctly
+    // if errors are returned from the API.
     // Normalize field-specific errors, if any.
-    const errors = fieldErrors.map((error) => ({
-      detail: error.description,
-      // Going forward, nested attributes will be hoisted.  Thus "attributes."
-      // must be stripped from the error name key.  Until all models are
-      // transitioned away from model fragments, however, both this and the
-      // prior error key mappings are necessary.
-      source: {
-        pointer: `/data/attributes/${error.name.replace('attributes.', '')}`,
-      },
-    }));
+    const errors = fieldErrors.map((error) => {
+      if (error.name === 'attributes.filter') {
+        error.name = 'attributes.filter_string';
+      }
```

Comment:
I took a look at the ember API and I think it might make more sense to have the custom logic go through the [extractErrors](https://api.emberjs.com/ember-data/4.12/classes/RESTSerializer/methods/extractErrors?anchor=extractErrors) hook

Reply:
Updated. Much nicer now with extractErrors